src/toolbar/ToolBar.js

```diff
@@ -215,7 +215,13 @@ class ToolBar extends Component {
     }
 
     const searchTextInput = this.renderSearchPanel();
-    const modal = this.props.enableInsert ? this.renderInsertRowModal() : null;
+    let modal = this.props.enableInsert ? this.renderInsertRowModal() : null;
+
+    if (this.props.enableInsert) {
```

Comment:
we just follow the bootstrap, esc will trigger close event for modal at default. If I've free time, I'll make it as configurable.